src/fuel-vm/instruction-set.md

```diff
@@ -1796,33 +1796,35 @@ Panic if:
 - `$rB + 32` overflows or `> VM_MAX_RAM`
 - Contract with ID `MEM[$rB, 32]` is not in `tx.inputs`
 
-### `LDC`: Load code from an external contract or blob
+### `LDC`: Load code from an external contract, blob or memory
 
 |             |                                                                                                                                                   |
 |-------------|---------------------------------------------------------------------------------------------------------------------------------------------------|
-| Description | Copy `$rC` bytes of code at offset `$rB` from object with 32 byte id starting at `$rA` into memory starting at `$ssp`. Object type is in `imm`.   |
-| Operation   | `id = mem[$rA,32]; code = match imm { 0 => contract_code($id), 1 => blob_payload($id) }; MEM[$ssp, $rC] = code[$rB, $rC];`                        |
+| Description | Copy `$rC` bytes of code at offset `$rB` from object identified with `$rA` into memory starting at `$ssp`. Object type is in `imm`.               |
+| Operation   | `code = match imm { 0 => contract_code(mem[$rA,32]), 1 => blob_payload(mem[$rA,32]), 2 => mem[$ra, ..] }; MEM[$ssp, $rC] = code[$rB, $rC];`       |
 | Syntax      | `ldc $rA, $rB, $rC, imm`                                                                                                                          |
 | Encoding    | `0x00 rA rB rC imm`                                                                                                                               |
 | Notes       | If `$rC` is greater than the code size, zero bytes are filled in.                                                                                 |
 
-Object type from `imm` determined the source for loading as follows:
+Object type from `imm` determines the source for loading as follows:
 
 | `imm` | Object type   |
 |-------|---------------|
 | `0`   | Contract code |
 | `1`   | Blob payload  |
+| `2`   | VM memory     |
 | other | _reserved_    |
 
 Panic if:
 
 - `$ssp + $rC` overflows or `> VM_MAX_RAM`
-- `$rA + 32` overflows or `> VM_MAX_RAM`
+- `imm <= 1` and `$rA + 32` overflows or `> VM_MAX_RAM`
 - `$ssp + $rC >= $hp`
 - `imm == 0` and `$rC > CONTRACT_MAX_SIZE`
 - `imm == 0` and contract with ID `MEM[$rA, 32]` is not in `tx.inputs`
```

Comment:
Do we want to hi light that we return an error if it is called in the context of predicates?

Reply:
It's already called out in the predicate section of https://github.com/FuelLabs/fuel-specs/pull/612, but we could do that.

Reply:
e4ee95b01c286d58230c0c89e621ffe685dc1165